### ISSUE TITLE
Project Update 1

### ISSUE THREAD

**Issue by XiaoxuanLu** [XiaoxuanLu/Movie_recommender_system#1]
# Project Update 1

Until now, we split into two groups to build two separate recommendation systems. Yiming and Xiaoyu are building a simple recommender system, and Violette and Xiaoxuan are working on building a content-based recommender system. 

## Individual work so far:

Violette: Started looking at the datasets for content-based recommender systems and visualising them with graphs. Researched methods for determining similarity between word embeddings with Xiaoxuan (see below).

Xiaoxuan: Preprocessed the dataset only to keep the valuable information and create a new feature for them. Tokenized and Lemmatized the new feature using NLTK and regular expression. Researched how to find text similarities with Violette and decided to use TF-IDF cosine similarity, KL Divergence, and clustering methods to find the most ten similar movies. Other possible measures include Euclidean distance and adjusted cosine similarity.

Xiaoyu: made a simple recommender that uses the IMDB formula to generate each movie’s ranking and then sorts movies using this ranking mechanism. Also, I have included several filters to assist users in searching for top-rated based on their preferences. The basic chart is designed to be genre-specific. Then additional features like the year and the language of the movie are added to the original chart to accommodate the users better. Besides, I have also developed a novel filter that takes account of the runtime of the movie, allowing users to select movies that best fit into their schedule. (all codes are in Simple-recommender notebook) 

Yiming: continued on feature additions of the simple recommender like ranking on movie popularity and filtering with names of directors and cast members. Researched approaches to building the dedicated website for our movie recommender system.

## Goals for next week?
Finish visualising data and decide what content to use for the movie recommender.
Start building the content-based recommender system using several different ways. 

## Questions?

How to evaluate the effectiveness of our recommender system? We considered comparing our results to Netflix's recommendations, but we are clearly using different datasets. 

How do you combine three different text similarity methods? For example, we could get the ten most similar movies for each technique, and how could we average the 30 movies to ensure it works efficiently.

**Comment by XiaoxuanLu** [comment]
To some extent, you are already evaluating the systems by comparing them (to a more simple baseline). If your dataset does not have a gold standard for recommendation it can indeed be difficult to evaluate. Is there any way to use the dataset? For example, split it into a training set and test set and then see if your system makes recommendations that were highly rated by those users in the test set? If there is no such possibility, maybe you can do a simple manual evaluation yourself? You can also try to evaluate some partial aspects automatically like the consistency of the recommendations, if this seems relevant. It is indeed possible to combine different similarity methods! This is sometimes done if the three have different benefits. You could try computing a (weighted?) average of different similarity methods and then taking the top 10 according to that combined metric.
Jelke Bloem, 12 May at 0:20
